From 2669780bfa4e462e1e999b87175d361556e871c7 Mon Sep 17 00:00:00 2001
From: Yuya Hamamachi <yuya.hamamachi.sx@renesas.com>
Date: Tue, 10 Oct 2023 20:28:08 +0900
Subject: [PATCH 1/4] HACK: ufs: Force enable initilized flag

If UFS has already initilaized on U-Boot, initilization fails on Linux
driver.
However, currently there is no mechaniseme to chacking whether UFS is
already initilized or not.
Thus, this patch make to enable initialized flag forcely.

Signed-off-by: Yuya Hamamachi <yuya.hamamachi.sx@renesas.com>
---
 drivers/scsi/ufs/ufs-renesas.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/scsi/ufs/ufs-renesas.c b/drivers/scsi/ufs/ufs-renesas.c
index 4b26f1f36b7c..1e662513352a 100644
--- a/drivers/scsi/ufs/ufs-renesas.c
+++ b/drivers/scsi/ufs/ufs-renesas.c
@@ -589,6 +589,8 @@ static void ufs_renesas_pre_init(struct ufs_hba *hba)
 	struct ufs_renesas_priv *priv = ufshcd_get_variant(hba);
 	const struct soc_device_attribute *attr;
 
+	priv->initialized = true;
+
 	if (priv->initialized)
 		return;
 
-- 
2.34.1

